crates/integrations/datafusion/src/schema.rs

```diff
@@ -89,14 +87,43 @@ impl SchemaProvider for IcebergSchemaProvider {
     }
 
     fn table_names(&self) -> Vec<String> {
-        self.tables.keys().cloned().collect()
+        self.tables
+            .keys()
+            .flat_map(|table_name| {
+                [table_name.clone()]
+                    .into_iter()
+                    .chain(MetadataTableType::all_types().map(|metadata_table_name| {
+                        format!("{}${}", table_name.clone(), metadata_table_name.as_str())
```

Comment:
The `$` syntax follows Flink https://iceberg.apache.org/docs/1.5.2/flink-queries/#inspecting-tables instead of `.` from Spark, because 4-level `.` is not supported in Datafusion.